client/src/pages/SignUp.js

```diff
@@ -0,0 +1,181 @@
+import {
+  TextField,
+  Button,
+  Typography,
+  Link,
+  makeStyles,
+  Paper,
+  Grid,
+  Box,
+} from '@material-ui/core';
+import React, { useCallback, useEffect, useState } from 'react';
+
+const useStyles = makeStyles({
+  textField: {
+    width: '50%',
+  },
+  bold: {
+    color: 'black',
+  },
+  seperator: {
+    width: '10%',
+    border: '1px solid #00e676',
+  },
+  formPadding: {
+    padding: '20px',
+  },
+});
+
+const initialFormData = {
+  registerEmail: '',
+  registerName: '',
+  registerPassword: '',
+  confirmPassword: '',
+};
+
+const initialValData = {
+  lengthValidation: true,
+  confirmValidation: true,
+};
+
+const SignUp = () => {
+  const [formData, setFormData] = useState(initialFormData);
+  const [valData, setValData] = useState(initialValData);
+
+  const classes = useStyles();
+  //change this to change the min length requirement
+  const minLength = 6;
+  const handleValidation = useCallback(() => {
+    setValData((prevState) => ({
+      ...prevState,
+      lengthValidation:
+        formData.registerPassword.length >= minLength ||
+        formData.registerPassword.length === 0,
+    }));
+    setValData((prevState) => ({
+      ...prevState,
+      confirmValidation: formData.registerPassword === formData.confirmPassword,
+    }));
```

Comment:
```suggestion
    setValData((prevState) => ({
      ...prevState,
      lengthValidation:
        formData.registerPassword.length >= minLength ||
        formData.registerPassword.length === 0,
      confirmValidation: formData.registerPassword === formData.confirmPassword,
    }));
```
You can consolidate this